packages/frontend/src/redux/actions/account.js

```diff
@@ -505,6 +518,19 @@ export const {
             }
         },
         () => showAlert()
+    ],
+    ADD_SHARD_ACCESS_KEY: [
+        async (accountId, contractId, publicKey, fullAccess = false, methodNames = '', shardInfo) => {
+            try {
+                const account = await wallet.getAccount(accountId);
+                const signature = await wallet.signatureFor(account);
+                await syncPrivateShardAccount({ accountId, publicKey, signature, shardInfo });
+                await new Wallet(shardInfo).addAccessKey(accountId, contractId, publicKey, fullAccess, methodNames);
```

Comment:
We send a sync account request to the calimero endpoint. If the signature is correct a new account with the same name and public keys is created in the shard. After this a wallet pointing to the shard RPC is used to add the new access key.